crawler.py

```diff
@@ -0,0 +1,54 @@
+from selenium import webdriver
+from bs4 import BeautifulSoup, SoupStrainer
+import unicodedata
+import re
+import traceback
+
+
+class Crawler:
+
+    def __init__(self, url):
+        self.url = url
+        self.queue = []
+        self.res = set()
+        self.email_regex = re.compile('([A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4})', re.IGNORECASE)
+
+    def connect(self):
+        driver = webdriver.Firefox()
+        driver.get("http://" + self.url)
+        html = driver.page_source
+        soup = BeautifulSoup(html)
+        for a in soup.find_all('a', href=True):
```

Comment:
This seems like it is only looking at <a> tags on the main page, but no links from other pages within the same domain. Basically it seems like it's assuming that all pages within the site are accessible from the main page.